Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs

```diff
@@ -0,0 +1,205 @@
+            interpolator.SetTargetPosition(placementPosition);
+
+            // Rotate this object to face the user.
+            interpolator.SetTargetRotation(Quaternion.Euler(0, Camera.main.transform.localEulerAngles.y, 0));
+        }
+
+        public virtual void OnInputClicked(InputClickedEventData eventData)
+        {
+            // On each tap gesture, toggle whether the user is in placing mode.
+            IsBeingPlaced = !IsBeingPlaced;
+            HandlePlacement();
+        }
+
+        private void HandlePlacement()
+        {
+            if (IsBeingPlaced)
+            {
+                SetLayerRecursively(transform, useDefaultLayer: false);
+                InputManager.Instance.AddGlobalListener(gameObject);
+
+                // If the user is in placing mode, display the spatial mapping mesh.
+                SpatialMappingManager.Instance.DrawVisualMeshes = true;
+#if UNITY_WSA && !UNITY_EDITOR
+
+                //Removes existing world anchor if any exist.
+                WorldAnchorManager.Instance.RemoveAnchor(gameObject);
+#endif
+            }
+            else
+            {
+                SetLayerRecursively(transform, useDefaultLayer: true);
+                // Clear our cache in case we added or removed gameobjects between taps
+                defaultLayersCache.Clear();
+                InputManager.Instance.RemoveGlobalListener(gameObject);
+
+                // If the user is not in placing mode, hide the spatial mapping mesh.
+                SpatialMappingManager.Instance.DrawVisualMeshes = false;
+#if UNITY_WSA && !UNITY_EDITOR
+
+                // Add world anchor when object placement is done.
+                WorldAnchorManager.Instance.AttachAnchor(gameObject, SavedAnchorFriendlyName);
+#endif
+            }
+        }
+
+        private void DetermineParent()
+        {
+            if (!PlaceParentOnTap) { return; }
+
+            if (ParentGameObjectToPlace == null)
+            {
+                if (gameObject.transform.parent == null)
+                {
+                    Debug.LogWarning("The selected GameObject has no parent.");
+                    PlaceParentOnTap = false;
+                }
+                else
+                {
+                    Debug.LogWarning("No parent specified. Using immediate parent instead: " + gameObject.transform.parent.gameObject.name);
+                    ParentGameObjectToPlace = gameObject.transform.parent.gameObject;
+                }
+            }
+
+            if (ParentGameObjectToPlace != null && !gameObject.transform.IsChildOf(ParentGameObjectToPlace.transform))
+            {
+                Debug.LogWarning("The specified parent object is not a parent of this object.");
+            }
+        }
+
+        private static void SetLayerRecursively(Transform objectToSet, bool useDefaultLayer)
+        {
+            if (useDefaultLayer)
+            {
+                if (defaultLayersCache.ContainsKey(objectToSet.gameObject))
```

Comment:
Because there's no real way to keep track of any children that get added or removed.